pipeline/transfer.py

```diff
@@ -150,37 +156,48 @@ def transfer(args):
 
     for filter_tag in ["filtered", "unfiltered"]:
         couplings_nobeam = {}
-        for spin_pair in spin_pairs:
-            if filter_tag == "filtered":
-                tbmcm = np.einsum('ijk,jklm->iklm', trans[spin_pair],
-                                  mcms_dict_nobeam[spin_pair])
-            else:
-                tbmcm = mcms_dict_nobeam[spin_pair]
-            btbmcm = np.transpose(
-                np.array([np.sum(
-                    tbmcm[:, :, :, nmt_binning.get_ell_list(i)], axis=-1)
-                          for i in range(n_bins)]),
-                axes=[1, 2, 3, 0])
-            # Invert and multiply by tbmcm to get final bandpower
-            # window functions.
-            if spin_pair == "spin0xspin0":
-                size = 1
-            elif spin_pair == "spin0xspin2":
-                size = 2
-            elif spin_pair == "spin2xspin2":
-                size = 4
-            ibtbmcm = np.linalg.inv(btbmcm.reshape([size*n_bins, size*n_bins]))
-            winflat = np.dot(ibtbmcm, tbmcm.reshape([size*n_bins, size*nl]))
-            wcal_inv = ibtbmcm.reshape([size, n_bins, size, n_bins])
-            bpw_windows = winflat.reshape([size, n_bins, size, nl])
-
-            couplings_nobeam[f"bp_win_{spin_pair}"] = bpw_windows
-            couplings_nobeam[f"inv_coupling_{spin_pair}"] = wcal_inv
+        couplings_nobeam_pure = {}
+        for couplings, mcms_dict in zip([couplings_nobeam,
+                                         couplings_nobeam_pure],
+                                        [mcms_dict_nobeam,
+                                         mcms_dict_nobeam_pure]):
```

Comment:
This for loop should add something like:
```python
if not mcms_dict:
    # mcms_dict is empty
    # this means we are not using B-purify
    # and mcms_dict_nobeam_pure is empty
    continue
```
to handle the case not using B-purify. This empty dictionary detection is based on my previous comment.